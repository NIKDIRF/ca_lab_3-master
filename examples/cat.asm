.data:
    num tmp 0
.text:

loop: in tmp
    out_char tmp
	jmp loop
end: hlt