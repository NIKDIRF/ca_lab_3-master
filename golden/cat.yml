source: |
  .data:
      num tmp 0
  .text:

  loop: in tmp
      out_char tmp
  	jmp loop
  end: hlt
input: |
  Good news, everyone!

data:
- '0'
code:
- opcode: in
  addr: 0
  args:
  - 0
- opcode: out_char
  addr: 1
  args:
  - 0
- opcode: jmp
  addr: 2
  args:
  - 0
- opcode: hlt
  addr: 3
  args: []
output: |
  Good news, everyone!
